code-challenges.js

```diff
@@ -3,12 +3,21 @@
 
 // --------------------1) Create a function that takes in an array of objects and returns a sentence about each person with their names capitalized.
 
-var people = [{name: "ford prefect", occupation: "hitchhiker"}, {name: "zaphod beeblebrox", occupation: "president of the galaxy,"}, {name: "arthur dent", occupation: "radio employee"}]
+var people = [{name: "ford prefect", occupation: "hitchhiker"}, {name: "zaphod beeblebrox", occupation: "president of the galaxy"}, {name: "arthur dent", occupation: "radio employee"}]
 // Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is the president of the galaxy." "Arthus Dent is a radio employee."
 
 
+// I created a function that takes array as the argument and returns the expected output using string interpolation. I know this code doesn't look so pretty but it's the best I could come up with on my own without much googling. 
+// Essentially, I used .replace on all of the array names along with regex. In regex, the \b asserts a position at the beginning of all word boundries and \w selects the first word character (\w is used in conjunction with \b, otherwise, without \b, the \w would select all word characters instead of the first word character of each word).
+// The /g is a global search which means it will match all word occurrences within a string. After the regex, I set a variable c to equal .toUpperCase which is what uppercases all of the first characters of each first and last name in the array.
 
 
+const namesCapitalized = (array) => {
+        return (`${array[0].name.replace(/\b\w/g, c => c.toUpperCase())} is a ${array[0].occupation}.\n${array[1].name.replace(/\b\w/g, c => c.toUpperCase())} is the ${array[1].occupation}.\n${array[2].name.replace(/\b\w/g, c => c.toUpperCase())} is a ${array[2].occupation}.`) 
+    
+}
+console.log(namesCapitalized(people))
```

Comment:
good work! how can this code be refactored to anticipate any array? right now it works perfectly for the test array we have given you, but any change to that array would give you an unexpected return